test/server/drain_manager_impl_test.cc

```diff
@@ -37,12 +40,15 @@ TEST(DrainManagerImplTest, All) {
   // Test drain sequence.
   Event::MockTimer* drain_timer = new Event::MockTimer(&server.dispatcher_);
```

Comment:
It doesn't leak. This is a pretty common pattern in tests where code needs to operate on a unique_ptr but we need access to the returned thing so we have to the raw pointer. (This timer gets wrapped in a unique_ptr and returned to code under test).